packages/sanity/src/core/config/types.ts

```diff
@@ -388,6 +388,18 @@ export interface PluginOptions {
      */
     enableLegacySearch?: boolean
   }
+  /** allows for in studio feature toggles */
+  features?: {
+    /** allows for shorter-lived feature toggles */
+    beta?: {
+      /**
+       * @hidden
+       * @beta */
+      treeArrayEditing?: {
+        enabled: boolean
+      }
+    }
+  }
```

Comment:
Suggestion:  move this to its own interface (e.g, `Features`) since it is used in both the `PluginOptions` and `Source` interface

```ts
interface Features {
  /**
   * Configuration for beta features.
   */
  beta?: {
    /**
     * @beta
     * @hidden
     * */
    treeArrayEditing?: {
      /**
       * Enables the tree array editing feature.
       */
      enabled: boolean
    }
  }
}
```

Which are then used in the `PluginOptions` and `Source` interface:

```ts

/** @beta */
export interface Source {
  // rest of types ...

  /**  Configuration for various features. */
  features?: Features
}

/** @beta */
export interface PluginOptions {
  // rest of types ...

  /**  Configuration for various features. */
  features?: Features
}
```

Reply:
I also made some small updates to the TSDoc copy – what do you think?  :)

Reply:
Great suggestion, I've updated it :)